iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputFormat.java

```diff
@@ -83,9 +83,20 @@ private static HiveIcebergRecordWriter writer(JobConf jc) {
         .operationId(operationId)
         .build();
     String tableName = jc.get(Catalogs.NAME);
-    HiveFileWriterFactory hfwf = new HiveFileWriterFactory(table, fileFormat, schema,
-        null, fileFormat, null, null, null, null);
-    return new HiveIcebergRecordWriter(schema, spec, fileFormat,
-        hfwf, outputFileFactory, io, targetFileSize, taskAttemptID, tableName);
+    HiveFileWriterFactory writerFactory = new HiveFileWriterFactory(table, fileFormat, schema, null, fileFormat,
+        null, null, null, getPositionDeleteRowSchema(schema, fileFormat));
+    if (HiveIcebergStorageHandler.isDelete(jc, tableName)) {
+      return new HiveIcebergDeleteWriter(schema, spec, fileFormat, writerFactory, outputFileFactory, io, targetFileSize,
+          taskAttemptID, tableName);
+    } else {
+      return new HiveIcebergRecordWriter(schema, spec, fileFormat, writerFactory, outputFileFactory, io, targetFileSize,
+          taskAttemptID, tableName);
+    }
+  }
+
+  private static Schema getPositionDeleteRowSchema(Schema schema, FileFormat fileFormat) {
+    // TODO: remove this Avro-specific logic once we have Avro writer function ready
```

Comment:
Is it implemented in the Iceberg project? Is there an existing PR or issue for it?

Reply:
Actually we no longer need an exception for Avro, we can use the `DataWriter` class from the iceberg-avro module